backbone.js

```diff
@@ -980,11 +980,13 @@
       return slice.apply(this.models, arguments);
     },
 
-    // Get a model from the set by id.
+    // Get a model from the set by id, cid, model object with id or cid
+    // properties, or an attributes object that is transformed through modelId.
     get: function(obj) {
       if (obj == null) return void 0;
-      var id = this.modelId(this._isModel(obj) ? obj.attributes : obj);
-      return this._byId[obj] || this._byId[id] || this._byId[obj.cid];
+      return this._byId[obj] ||
+        this._byId[this.modelId(obj.attributes || obj)] ||
+        obj.cid && this._byId[obj.cid] || void 0;
```

Comment:
I think this is fine without the final guard. A 0 cid might be a worthwhile case to support, but I'd argue you're almost always going to be getting those `cids` from *somewhere* (i.e. an existing model) which [can never have a cid of 0](https://github.com/jashkenas/underscore/blob/1.8.3/underscore.js#L1373-L1379).